File: languages/pascal/prime.pas
program PrimeCheck;

uses sysutils;

var
   n: LongInt;
   isPrime: Boolean;
   i: LongInt;
begin
   Write('Enter a number to check if it''s prime: ');
   ReadLn(n);
   isPrime := True;

   if n <= 1 then
      isPrime := False
   else if n <= 3 then
      isPrime := True
   else if (n mod 2 = 0) or (n mod 3 = 0) then
      isPrime := False
   else
   begin
      i := 5;
      while i * i <= n do
      begin
         if (n mod i = 0) or (n mod (i + 2) = 0) then
         begin
            isPrime := False;
            Break;
         end;
         i := i + 6;
      end;
   end;

   if isPrime then
      WriteLn(n, ' is a prime number.')
   else
      WriteLn(n, ' is not a prime number.');
end.

File: languages/ocaml/prime.ml
let is_prime n =
    let rec check_from d =
        d * d > n || (n mod d <> 0 && check_from (d + 1))
    in
    n > 1 && check_from 2

let () =
    print_endline "Enter a number:";
    let n = read_int () in
    if is_prime n then
        Printf.printf "%d is a prime number.\n" n
    else
        Printf.printf "%d is not a prime number.\n" n
